### DISCUSSION TITLE
Hot and cold repo?

### DISCUSSION THREAD

**Discussion by KamikazeePL** [rustic-rs/rustic#733] [Q&A] [answered]
Could somebody explain me hot and cold repo terms.
The best answer would be with examples.

**Comment by aawsome** [comment, marked as answer]
see e.g. https://github.com/rustic-rs/rustic/blob/main/config/ovh-hot-cold.toml which is basically what I am using for my personal backups.

`rclone:ovh:backup-home` in this case points to a rclone config which uses OVH Cloud Archive, see https://www.ovhcloud.com/en/public-cloud/cloud-archive/. Note that this is a cheap storage but has access restrictions and also charges for access. Basically a typical example for what I would call cold storage.

`rclone:ovh:backup-home-hot` in this case points to a rclone config which uses OVH Object Storage, see https://www.ovhcloud.com/en/public-cloud/object-storage/. This is a standard object storage which allows easy direkt access, like AWS S3.

For OVH Cloud Archive, if I try to access an object, it will fail but immedately warm up the object. When I retry after the warm-up-time, the object is directly accessible by rclone. So I set `warm-up = true` and specify a reasonable warm-up time. 

That's basically all. Now the commands which really need to access the cold storage will automatically warm-up the needed objects and wait for them before proceeding. Note that all metadata is also saved in the hot repo, so many commands won't even need a warm-up but work as usual.

You can see statistics of the cold and hot part using `repoinfo` and `check` will also check consistency of hot/cold repository. In any case, the cold part contains a full and normal repository, so you can always just transfer that to a "normal" storage and keep accessing/using this as a normal repository from that time on.

**Reply by simonsan** [reply]
We should add this to the [FAQ](https://github.com/rustic-rs/rustic/blob/main/docs/FAQ.md) I feel!
